RealmSwift/Map.swift

```diff
@@ -779,6 +779,14 @@ public final class Map<Key: _MapKey, Value: RealmCollectionValue>: RLMSwiftColle
     }
 }
 
+// MARK: - Hashable
+
+extension Map: Hashable {
```

Comment:
Similarly, this is not at all a valid Hashable implementation.

Reply:
Using ObjectIdentifier given that we already conform to Equatable

Reply:
ObjectIdentifier is not a valid definition either. Two managed dictionary objects which correspond to the same field are equal to each other but have different ObjectIdentifiers. All of our accessor types are fundamentally incompatible with Hashable due to that their identity changes when you promote an unmanaged object.

Reply:
@tgoyne we do need to conform to `Hashable` to be able to include Map or List as a an associated value in AnyRealmValue if we want to go that approach.
I can think of a few approaches for this:
* Use ObjectIdentifer and sacrifice equality for managed and unmanaged object
* Have associated values as Swift's native types Array/Dictionary, this will clearly will go bad with nested collections which will need to be instantiated completely to convert it (Discarded)
* Store some reference to the collection as the associated value and then and have some static function that returns the collection (Not need to conform to hashable, but worse developer experience).

I'm open to discuss different solutions

Reply:
We can override the hash function on AnyRealmValue and return a constant value for the Map and List cases.